// bold-ui/lib/components/Select/SelectMenu/SelectMenuItem.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styles_1 = require("../../../styles");
var Spinner_1 = require("../../Spinner");
function SelectMenuItem(props) {
    var style = props.style, selected = props.selected, highlighted = props.highlighted, rest = __rest(props, ["style", "selected", "highlighted"]);
    var _a = styles_1.useStyles(exports.createStyles, props), classes = _a.classes, css = _a.css;
    return react_1.default.createElement("li", __assign({ className: css(classes.item, style) }, rest));
}
exports.SelectMenuItem = SelectMenuItem;
function SelectHelperMenuItem(props) {
    var theme = styles_1.useTheme();
    return react_1.default.createElement(SelectMenuItem, __assign({ style: { background: theme.pallete.surface.background } }, props));
}
exports.SelectHelperMenuItem = SelectHelperMenuItem;
function SelectLoadingItem(props) {
    var theme = styles_1.useTheme();
    return (react_1.default.createElement(SelectHelperMenuItem, __assign({}, props),
        "Carregando...",
        react_1.default.createElement(Spinner_1.Spinner, { style: { color: theme.pallete.primary.main, float: 'right' } })));
}
exports.SelectLoadingItem = SelectLoadingItem;
function SelectEmptyItem(props) {
    return react_1.default.createElement(SelectHelperMenuItem, __assign({}, props), "Nenhum item encontrado");
}
exports.SelectEmptyItem = SelectEmptyItem;
exports.createStyles = function (theme, _a) {
    var highlighted = _a.highlighted;
    return ({
        item: {
            borderBottom: "1px solid " + theme.pallete.divider,
            cursor: 'pointer',
            padding: '0.325rem 0.5rem',
            transition: '.1s ease',
            background: highlighted && theme.pallete.surface.background,
            '&:last-of-type': {
                borderBottom: 'none',
            },
            '&:hover': {
                background: theme.pallete.surface.background,
            },
        },
    });
};
