// bold-ui/lib/styles/theme/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var createBreakpoints_1 = require("./createBreakpoints");
exports.breakpoints = createBreakpoints_1.breakpoints;
exports.createBreakpoints = createBreakpoints_1.createBreakpoints;
var createPallete_1 = require("./createPallete");
exports.createPallete = createPallete_1.createPallete;
var createTypography_1 = require("./createTypography");
exports.createTypography = createTypography_1.createTypography;
var createShadows_1 = require("./createShadows");
exports.createShadows = createShadows_1.createShadows;
var radius_1 = require("./radius");
exports.radius = radius_1.radius;
var ThemeContext_1 = require("./ThemeContext");
exports.ThemeContext = ThemeContext_1.ThemeContext;
exports.ThemeProvider = ThemeContext_1.ThemeProvider;
var themes_1 = require("./themes");
exports.lightTheme = themes_1.lightTheme;
exports.darkTheme = themes_1.darkTheme;
var zIndex_1 = require("./zIndex");
exports.zIndex = zIndex_1.zIndex;
var createTheme_1 = require("./createTheme");
exports.createTheme = createTheme_1.createTheme;
exports.focusBoxShadow = createTheme_1.focusBoxShadow;
