// bold-ui/lib/components/Tooltip/Tooltip.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var usePopper_1 = __importDefault(require("../../hooks/usePopper"));
var styles_1 = require("../../styles");
var string_1 = require("../../util/string");
var Portal_1 = require("../Portal");
var RootRef_1 = require("../RootRef");
var FadeTransition_1 = require("../Transition/FadeTransition");
var TooltipPopper_1 = require("./TooltipPopper");
function Tooltip(props) {
    var text = props.text, children = props.children, offset = props.offset, externalStyle = props.style, container = props.container, rest = __rest(props, ["text", "children", "offset", "style", "container"]);
    var child = react_1.default.Children.only(children);
    var _a = styles_1.useStyles(createStyles), theme = _a.theme, css = _a.css, classes = _a.classes;
    var _b = react_1.useState(false), visible = _b[0], setVisible = _b[1];
    var tooltipIdRef = react_1.useRef("tooltip-" + string_1.randomStr());
    var anchorRef = react_1.useRef();
    var tooltipRef = react_1.useRef();
    var _c = usePopper_1.default(__assign({ anchorRef: anchorRef, popperRef: tooltipRef, modifiers: {
            offset: { offset: "0, " + theme.typography.sizes.html * offset },
            preventOverflow: {
                boundariesElement: 'window',
            },
        } }, rest), [visible]), popperStyle = _c.style, placement = _c.placement;
    react_1.useEffect(function () {
        if (!anchorRef.current || !visible) {
            return;
        }
        var handleWindowMouseOver = function (e) {
            // This is implemented using mouseover since mouseleave does not trigger
            // for disabled elements due to browser/react bugs (https://github.com/facebook/react/issues/4251)
            var target = e.target;
            if (!anchorRef.current.contains(target)) {
                setVisible(false);
            }
        };
        window.addEventListener('mouseover', handleWindowMouseOver);
        return function () { return window.removeEventListener('mouseover', handleWindowMouseOver); };
    }, [anchorRef.current, visible]);
    var handleMouseEnter = function (e) {
        setVisible(true);
        child.props.onMouseEnter && child.props.onMouseEnter(e);
    };
    var handleFocus = function (e) {
        setVisible(true);
        child.props.onFocus && child.props.onFocus(e);
    };
    var handleBlur = function (e) {
        setVisible(false);
        child.props.onBlur && child.props.onBlur(e);
    };
    if (!text) {
        return child;
    }
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(RootRef_1.RootRef, { rootRef: anchorRef }, react_1.default.cloneElement(child, {
            'aria-describedby': tooltipIdRef.current,
            onMouseEnter: handleMouseEnter,
            onFocus: handleFocus,
            onBlur: handleBlur,
        })),
        react_1.default.createElement(FadeTransition_1.FadeTransition, { in: visible }, function (_a) {
            var className = _a.className;
            return (react_1.default.createElement(Portal_1.Portal, { container: container },
                react_1.default.createElement("div", { id: tooltipIdRef.current, ref: tooltipRef, className: css(className, popperStyle, classes.popper, visible && classes.shown), role: 'tooltip', "aria-hidden": !visible ? 'true' : 'false', "data-placement": placement },
                    react_1.default.createElement(TooltipPopper_1.TooltipPopper, { text: text, style: externalStyle }))));
        })));
}
exports.Tooltip = Tooltip;
Tooltip.defaultProps = {
    placement: 'top',
    offset: 0.25,
};
var createStyles = function (theme) { return ({
    popper: {
        zIndex: theme.zIndex.tooltip,
        display: 'none',
    },
    shown: {
        display: 'block',
    },
}); };
