// bold-ui/lib/components/Select/SelectSingle/SelectDownshiftMenu.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styles_1 = require("../../../styles");
var SelectMenu_1 = require("../SelectMenu");
var SelectMenuItem_1 = require("../SelectMenu/SelectMenuItem");
function SelectDownshiftMenu(props) {
    var items = props.items, isLoading = props.loading, components = props.components, createNewItem = props.createNewItem, _a = props.downshift, isOpen = _a.isOpen, getMenuProps = _a.getMenuProps;
    var classes = styles_1.useStyles(exports.createStyles).classes;
    var _b = __assign(__assign({}, exports.defaultComponents), components), CreateItem = _b.CreateItem, LoadingItem = _b.LoadingItem, EmptyItem = _b.EmptyItem, Item = _b.Item;
    return (react_1.default.createElement("div", { className: classes.wrapper }, isOpen && (react_1.default.createElement(SelectMenu_1.SelectMenu, __assign({}, getMenuProps({ refKey: 'menuRef' })),
        isLoading && react_1.default.createElement(LoadingItem, __assign({}, props)),
        !isLoading && createNewItem && (items || []).length > 0 && react_1.default.createElement(CreateItem, __assign({}, props)),
        !isLoading && !createNewItem && (items || []).length === 0 && react_1.default.createElement(EmptyItem, __assign({}, props)),
        items && items.map(function (item, index) { return react_1.default.createElement(Item, __assign({ key: index, index: index, item: item }, props)); })))));
}
exports.SelectDownshiftMenu = SelectDownshiftMenu;
exports.defaultComponents = {
    CreateItem: function (props) { return react_1.default.createElement(SelectMenuItem_1.SelectCreateItem, null); },
    LoadingItem: function (props) { return react_1.default.createElement(SelectMenu_1.SelectLoadingItem, null); },
    EmptyItem: function (props) { return react_1.default.createElement(SelectMenu_1.SelectEmptyItem, null); },
    Item: function (props) {
        var renderItem = props.renderItem, item = props.item, index = props.index, _a = props.downshift, itemToString = _a.itemToString, highlightedIndex = _a.highlightedIndex, getItemProps = _a.getItemProps;
        return (react_1.default.createElement(SelectMenu_1.SelectMenuItem, __assign({ selected: highlightedIndex === index }, getItemProps({ item: item })), renderItem ? renderItem(item) : itemToString(item)));
    },
};
SelectDownshiftMenu.defaultProps = {
    components: exports.defaultComponents,
};
exports.createStyles = function () { return ({
    wrapper: {
        position: 'relative',
    },
}); };
