// bold-ui/lib/components/Popover/Popover.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styles_1 = require("../../styles");
var PopperFocus_1 = require("../Popper/PopperFocus");
function Popover(props) {
    var text = props.text, title = props.title, placement = props.placement, children = props.children;
    var renderPopper = function () {
        return (react_1.default.createElement(PopoverBase, { title: title },
            react_1.default.createElement("p", null, text)));
    };
    return (react_1.default.createElement(PopperFocus_1.PopperFocus, { renderPopper: renderPopper, placement: placement, offset: 0.75 }, children));
}
exports.Popover = Popover;
Popover.defaultProps = {
    text: '',
    title: null,
    placement: 'right',
};
function PopoverBase(props) {
    var title = props.title, children = props.children;
    var classes = styles_1.useStyles(exports.createStyles).classes;
    return (react_1.default.createElement("div", { className: classes.base },
        title && react_1.default.createElement("h4", { className: classes.title }, title),
        children));
}
exports.PopoverBase = PopoverBase;
exports.createStyles = function (theme) { return ({
    base: {
        borderRadius: theme.radius.popper,
        maxWidth: 300,
        background: theme.pallete.surface.main,
        boxShadow: "\n        0 1px 3px 0 rgba(0, 0, 0, 0.2),\n        0 2px 1px -1px rgba(0, 0, 0, 0.12),\n        0 1px 1px 0 rgba(0, 0, 0, 0.14)\n    ",
        padding: '1rem',
    },
    title: {
        textAlign: 'center',
        marginBottom: '0.5rem',
    },
}); };
